src/AuthenticatedAPIClient/AccessToken.js

```diff
@@ -0,0 +1,91 @@
+import Cookies from 'universal-cookie';
+import jwtDecode from 'jwt-decode';
+import axios from 'axios';
+import { logInfo, logError } from '@edx/frontend-logging';
+
+const httpClient = axios.create();
+const cookies = new Cookies();
+
+
+const getCookieValue = (name) => {
+  try {
+    return cookies.get(name);
+  } catch (error) {
+    logInfo(`Error reading the access token cookie: ${name}.`);
+    return null;
+  }
+};
+
+const decodeJwtCookie = (cookieValue) => {
+  try {
+    return jwtDecode(cookieValue);
+  } catch (error) {
+    logInfo('Error decoding JWT token.', {
+      jwtDecodeError: error,
+      cookieValue,
+    });
+    return null;
+  }
+};
+
+const isTokenExpired = token => !token || token.exp < Date.now() / 1000;
+
+
+export default class AccessToken {
+  constructor({ cookieName, refreshEndpoint, handleUnexpectedRefreshFailure }) {
+    this.cookieName = cookieName;
+    this.refreshEndpoint = refreshEndpoint;
+    this.handleUnexpectedRefreshFailure = handleUnexpectedRefreshFailure;
+
+    this.readJwtToken();
+
+    if (!this.value || this.isExpired) {
+      this.refresh();
+    }
+  }
+
+  readJwtToken() {
```

Comment:
Consider `loadJwtTokenFromCookie`.
* Load: because you are reading and caching.  Could also be `readAndCacheJwtTokenFromCookie`. :)
* FromCookie: just to clarify where it is coming from.